### ISSUE TITLE
BufferWriter.writeSlice() ignores Uint8Array length

### ISSUE THREAD

**Issue by cyraxx** [iwoplaza/typed-binary#35]
`BufferWriter.writeSlice()` calls `unwrapBuffer()` to get its own view of the underlying buffer. However, this only takes into account the `byteOffset` and not the `byteLength`. So it always creates its view all the way to the end of the underlying buffer.

Example to reproduce the issue:

```typescript
// Create a 2000 byte ArrayBuffer.
const someBuffer = new ArrayBuffer(2000);

// Create a 100 byte Uint8Array view into the ArrayBuffer at offset 10.
const someView = new Uint8Array(someBuffer, 10, 100);
console.log(`Should write ${someView.length} bytes`);

// Write the Uint8Array into a BufferWriter.
console.log(`Position before writing: ${output.currentByteOffset}`);
output.writeSlice(someView);
console.log(`Position after writing: ${output.currentByteOffset}`);
```

This will show that 1990 bytes have been written to the output instead of just 100.

**Comment by iwoplaza** [comment]
Thank you for reporting the issue! Definitely looks like this should not be happening.
I will get to fixing it promptly and include it as 4.1.1 patch release. I will also include your minimal reproduction as a test case so we do not see it reappear as a regression.

**Comment by iwoplaza** [comment]
Just included the fix in [v4.1.1](https://github.com/iwoplaza/typed-binary/releases/tag/v4.1.1), available via NPM.
Thanks again for the report! 💚
